tasks/init.rb

```diff
@@ -83,6 +83,11 @@ def request(method, uri, headers, body, opts)
 
   # Parses the response body.
   def parse_response_body(response, json)
+    # Per RFC2616:
+    # > All 1xx (informational), 204 (no content), and 304 (not modified)
+    # > responses MUST NOT include a message-body
+    return nil if [100..199, [204, 304]].any? { |x| x.include?(response.code.to_i) }
```

Comment:
```suggestion
    return nil unless response.read_body
```

Instead of skipping reading/parsing the body based on the response code, we should probably just skip reading/parsing if the response doesn't include a body at all since the task does not enforce proper formatting for the response.

Reply:
:+1: That's a really elegant way of resolving all the questions in the spec tests.

I'll commit this and then rip out all the tests we won't need.